File: frontend/src/Components/PopUpMessage.js
import React, {useState} from 'react'
import { Modal, Button } from 'react-bootstrap';

export default function PopUpMessage(props) {
    const [isVisible,setVisible] = useState(props.show ? props.show : false);

    function closeButton(event) {
        setVisible(false);
    }
    console.log("Popup:",props.title);

    return (
        <>
            <Modal name="popMsg" show={isVisible} centered size='lg'>
                <Modal.Header name="popTitle"closeButton>
                    <Modal.Title>{props.title}</Modal.Title>
                </Modal.Header>
                <Modal.Body name="popMessage">
                    {props.message}
                </Modal.Body>
                <Modal.Footer>
                    {props.closeButtonText ? <Button name="btnPopClose" onClick={closeButton}>{props.closeButtonText}</Button> : null}
                </Modal.Footer>
            </Modal>
        </>

    );
}

File: src/test/java/selenium/ComponentWrappers/PopUpMessage.java
package selenium.ComponentWrappers;

import org.openqa.selenium.By;

public abstract class PopUpMessage {
    static final public By byModal = By.name("popMsg");
    static final public By byTitle = By.name("popTitle");
    static final public By byBody = By.name("popMessage");
    static final public By byBtnClose = By.name("btnPopClose");
}
